plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java

```diff
@@ -711,24 +711,27 @@ public Optional<ProjectionApplicationResult<ConnectorTableHandle>> applyProjecti
                 false));
     }
 
-    private static boolean isPushDownSupported(ConnectorExpression connectorExpression)
+    private static boolean isSupportedForPushDown(ConnectorExpression connectorExpression)
     {
-        if (!(connectorExpression instanceof FieldDereference fieldDereference)) {
+        if (connectorExpression instanceof Variable) {
             return true;
         }
-        RowType rowType = (RowType) fieldDereference.getTarget().getType();
-        if (isDBRefField(rowType)) {
-            return false;
-        }
-        Field field = rowType.getFields().get(fieldDereference.getField());
-        if (field.getName().isEmpty()) {
-            return false;
-        }
-        String fieldName = field.getName().get();
-        if (fieldName.contains(".") || fieldName.contains("$")) {
-            return false;
+        else if (connectorExpression instanceof FieldDereference fieldDereference) {
```

Comment:
Nit: Can we replace `else if` with `if` - (the first `if` returns the value directly so we can have an independent `if` statement.